Mandate expiration of ClientInitKeys and presence of some extensions (#275)
mlswg/mls-protocol: draft-ietf-mls-protocol.md

CipherSuite supported_ciphersuites<0..255>;
~~~~~

> This extension MUST be always present in an MLS ClientInitKey.

Reviewer: There are two extensions here.  So if you mean both of them, you should say so.

> These extensions MUST be always present in an MLS ClientInitKey.

Reviewer 2: I believe this is only one extension

Reviewer 2: but "MUST always be" would be nicer

> These extensions MUST be always present in an MLS ClientInitKey.

Reviewer 2: ```suggestion
These extensions MUST be always present in a ClientInitKey.
```

uint64 expiration;
~~~~~

Applications that rely on "last resort" ClientInitKeys MAY set the
expiration to its maximum value even though this is NOT RECOMMENDED.
It is RECOMMENDED to rotate last resort keys at a pace chosen by the
application even though they can have much longer lifetimes than other

> ClientInitKeys.

Reviewer 2: The expiration of ClientInitKeys is important for PCS, so setting to the maximum value shouldn't be encouraged. If you have keys that never expire and you're compromised, those keys can be used at any time in the future to "add" you to a group when they're actually giving the adversary access.

Last Resort keys are a defense against DoS, when somebody sends you a lot of messages in a short period of time. As such, any ClientInitKey can be a Last Resort key, regardless of how long it's valid for, because nobody is checking how many times a specific key gets used.

Last Resort keys are not a suitable defense against the user being offline for an arbitrary amount of time. Handling this securely probably requires more consideration.

We should recommend an application-specified maximum duration, and switch to a `(notBefore, notAfter)` system so that everyone know's the key's lifetime.

> This extension MUST be always present in an MLS ClientInitKey.

Reviewer: The need for this is not clear to me.  Let's discuss this at the interim before merging this PR.

Reviewer: This was discussed at the interim, and people were comfortable with this requirement.

> This extension MUST be always present in an MLS ClientInitKey.

Reviewer 2: ```suggestion
This extension MUST always be present in a ClientInitKey.
```
